articles/Frontend/Components/calculator.md

```diff
@@ -0,0 +1,867 @@
+- Scientific notation using exponent part is allowed: `3.1415e10`.
+- Regular arithmetic operators are supported, with natural precedence: `+`, `-`,
+`*`, `/`, `!`. Operators can be either unary or binary, depending on their
+meaning.
+- Expressions can be wrapped in parenthesis: `3*(5+2)`.
+- Terms are separated by spaces, however regular operators and parenthesis also
+act as separators.
+- Functions and constants are supported: `cos PI`.
+- Variables are supported, but require to be defined: `3 * x`.
+- Any functions can be turned into a binary operator by simply prefixing them
+with a `@`: `4 @nthrt 16` is equivalent to `nthrt(4, 16)`. Obviously, only 
+functions that accept two parameters can work properly with this trick.
+
+#### Numbers
+The number representation library is using
+[arbitrary precision](https://en.wikipedia.org/wiki/Arbitrary-precision_arithmetic)
+to store and manipulate the numbers. The chosen one is
+[Decimal.js](https://github.com/MikeMcl/decimal.js/), which a good compromise
+regarding precision and performances. It exposes a comprehensive API to
+manipulate and compute decimal numbers.
+
+As the parser in use is not resilient against the round-off issue, every
+built-in operations are replaced by the equivalent provided by `Decimal.js`.
+That allows to support a better numerical precision, but that also means the
+result of the parser is now a `Decimal` object.
+
+To ensure the parser is still working well after having replaced its built-in
+computation functions, some wrappers have been added, to convert on the fly the
+data type when needed. The benefit is we can feed the parser with native types,
+and then the wrappers will do the magic to ensure all values are represented
+with a compatible format.
+
+#### Lexer
+Manipulating the expression requires to be able to recognize its elements, and
+to do so the calculator component is relying on a lexer, that is able to cut the
+expression into atomic and clearly identified parts.
+
+The chosen library is [moo](https://github.com/no-context/moo) which is a well
+supported and optimized tokenizer. It accepts rational expressions to define the
+patterns to recognize.
+
+It is worth to mention this tokenizer is not utilized by the parser to recognize
+the expression, as the parser is bringing its own way to do that. The purpose of
+this tokenizer, inside the calculator, is to bring the ability to identify the
+elements of the expression, in order to manipulates them. Among other things,
+this allows to:
+- display properly each element with respect to their meaning (mathematical
+symbols or special tokens)
+- navigate freely between elements, and add or remove them easily
+- apply some business logic on top of them (for instance change the sign of the
+current operand, or recognize exponents)  
+
+### mathsEvaluator
+The `mathsEvaluator` is a master piece inside the calculator, as it processes
+the computation of the expressions and returns the result. Internally it relies
```

Comment:
are you sure you want to say "master piece" as in "chef-d'œuvre"? How about "a central piece of code, it processes..."?